components/service/glean/docs/metrics/testing-metrics.md

```diff
@@ -0,0 +1,68 @@
+# Unit testing glean metrics
+
+In order to support unit testing inside of client applications using glean, a set of testing API 
+functions have been included.  The intent is to make glean easier to test 'out of the box' in any
+client application it may be used in.  These functions expose a way to inspect and validate recorded
+metric values within the client application but are restricted to test code only through visibility
+annotations (`@VisibleForTesting(otherwise = VisibleForTesting.NONE)`).
+
+## General test API method semantics
+
+To check if a value exists (i.e. it has been recorded), there is a `testHasValue(pingName: String)`
+function on each of the metric instances.
+
+Example:
+```kotlin
+GleanMetrics.Foo.UriCount.testHasValue()
+```
+
+To check the actual values, there is a `testGetValue(pingName: String)` function on each of the
+metric instances.  This function will return a datatype appropriate to the specific type of the
+metric it is being used with.
+
+Example:
+```kotlin
+GleanMetrics.Foo.UriCount.testGetValue()
+```
+
+These functions require a `pingName` arugument.  This is the name of the ping that the metric is
```

Comment:
Following up to above, can we make that a separate sub section, *Testing metrics for custom pings*?

Reply:
Below the `testClearAllData()` section probably though?